firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryMutationQueue.java

```diff
@@ -187,6 +187,11 @@ public MutationBatch getNextMutationBatchAfterBatchId(int batchId) {
     return queue.size() > index ? queue.get(index) : null;
   }
 
+  @Override
+  public int getLargestUnacknowledgedBatchId() {
+    return queue.size() == 0 ? 0 : nextBatchId - 1;
```

Comment:
Google style guide generally recommends using isEmpty(), since it always runs in O(1).

Reply:
Done.